### DISCUSSION TITLE
Adds support for debugging mode

### DISCUSSION THREAD

**Discussion by freekmurze** [spatie/wordpress-ray#11] [General]
@jeffreyvr does WordPress have a way to detect if it is running locally? Is there some kind of DEVELOPING or DEBUGGING constant that WordPress devs tend to use?

If so, I could make it so that Ray would be disabled if such flag is set to false.

**Comment by freekmurze** [comment]
@slaFFik maybe you have suggestions for the above as well?

**Reply by slaFFik** [reply]
@freekmurze 
Yes, it has a global `WP_DEBUG` constant, but it can be enabled on any environment: prod, staging, dev, local, etc. Similar to APP_DEBUG in Laravel.

In WordPress 5.5 new function and constants have been added that handle exactly environments. See this blog post: https://make.wordpress.org/core/2020/07/24/new-wp_get_environment_type-function-in-wordpress-5-5/

`wp_get_environment_type()` function to retrieve the value.
`define( 'WP_ENVIRONMENT_TYPE', 'staging' );` to set the value in the `wp-config.php` file of the WordPress site.

**Reply by slaFFik** [reply]
Here is official documentation: https://developer.wordpress.org/reference/functions/wp_get_environment_type/

**Reply by jeffreyvr** [reply]
What @slaFFik said ;)

**Reply by freekmurze** [reply]
I'm now disabling Ray by default in production environments like this

https://github.com/spatie/wordpress-ray/blob/master/src/Ray.php#L31-L33

To enable in a production environment you should use `ray()->enable()`

I'll document this soon.

**Reply by jeffreyvr** [reply]
Looking at recent issues (https://github.com/spatie/wordpress-ray/issues/22), maybe it should be documented in the readme.txt(and the docs) installation instructions, that definition of the WP_ENVIRONMENT_TYPE is required. Or maybe you where already planning on doing so @freekmurze?

Another thing to consider is, to only use `wp_get_environment_type()` when sure that the WP_ENVIRONMENT_TYPE constant is set.

This would mean assumed is that if the environment is not specifically set, it is a development/local environment.
